src/travel-details-screens/components/FlexibleTransportBookingDetails.tsx

```diff
@@ -0,0 +1,201 @@
+import {FlexibleTransport} from '@atb/assets/svg/color/images';
+import {CloseCircle} from '@atb/assets/svg/color/icons/actions';
+import {
+  useTranslation,
+  getTextForLanguage,
+  TripDetailsTexts,
+} from '@atb/translations';
+import {Leg} from '@atb/api/types/trips';
+import {
+  View,
+  TouchableOpacity,
+  Linking,
+  ScrollView,
+  Platform,
+} from 'react-native';
+import {StyleSheet, useTheme} from '@atb/theme';
+
+import {ThemeText} from '@atb/components/text';
+import {FlexibleTransportBookingOptions} from './FlexibleTransportBookingOptions';
+import {BottomSheetContainer} from '@atb/components/bottom-sheet';
+import {ScreenHeaderWithoutNavigation} from '@atb/components/screen-header';
+import {getPublicCodeFromLeg, getLegBookingIsAvailable} from '../utils';
+import {FlexibleTransportMessageBox} from './FlexibleTransportMessageBox';
+import {useNow} from '@atb/utils/use-now';
+import {useRemoteConfig} from '@atb/RemoteConfigContext';
+
+import {Dimensions} from 'react-native';
+import {useFirestoreConfiguration} from '@atb/configuration/FirestoreConfigurationContext';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+
+const {width, height} = Dimensions.get('window');
+const isSmallScreen = width < 320 || height < 568;
+
+type FlexibleTransportBookingDetailsProps = {
+  leg: Leg;
+  close: () => void;
+};
+
+export const FlexibleTransportBookingDetails: React.FC<
+  FlexibleTransportBookingDetailsProps
+> = ({leg, close}) => {
+  const {t, language} = useTranslation();
+  const style = useStyle();
+  const {theme} = useTheme();
+
+  const {flex_booking_number_of_days_available} = useRemoteConfig();
+  const {configurableLinks} = useFirestoreConfiguration();
+
+  const publicCode = getPublicCodeFromLeg(leg);
+
+  const now = useNow(2500);
+  const bookingIsAvailable = getLegBookingIsAvailable(
+    leg,
+    now,
+    flex_booking_number_of_days_available,
+  );
+
+  const {bottom: safeAreaBottom} = useSafeAreaInsets();
+
+  return (
+    <BottomSheetContainer maxHeightValue={0.83}>
+      <ScreenHeaderWithoutNavigation
+        title={t(
+          TripDetailsTexts.flexibleTransport.needsBookingWhatIsThisTitle(
+            publicCode,
+          ),
+        )}
+        color="background_1"
+        setFocusOnLoad={false}
+      />
+      <TouchableOpacity style={style.closeIconContainer} onPress={close}>
+        <CloseCircle height={theme.icon.size.large} />
+      </TouchableOpacity>
+
+      <ScrollView
+        style={[
+          style.contentContainer,
+          {
+            marginBottom: Math.max(safeAreaBottom, theme.spacings.medium),
```

Comment:
Not sure about these calculations in the UI, what does this fix?

Reply:
It is part of the bugfix for https://github.com/AtB-AS/mittatb-app/pull/3736#issuecomment-1655766138

Reply:
It adds this gray area at the bottom, and sets it to whatever is biggest of safeAreaBottom and theme.spacings.medium:
![safeArea](https://github.com/AtB-AS/mittatb-app/assets/134292729/ff7b93da-12b3-4dad-8494-eed63b32cf76)